add a couple tests

lucask42 opened the pull request.

johnuphoff commented on scripts/utils.js:
```diff
@@ -35,16 +35,30 @@ const validAruments = args => {
+      );
+    }
     process.exit(0);
+    return {};
```
I'm not sure this will return...

lucask42 commented:
Yeah, I'll take that out.  My mock 'process.exit' allows the code to
continue

On Sun, Sep 22, 2019, 3:27 PM J Savage Uphoff <notifications@github.com>
wrote:

> *@johnuphoff* commented on this pull request.
> ------------------------------
>
> In scripts/utils.js
> <https://github.com/johnuphoff/mailed-it/pull/2#discussion_r326922493>:
>
> >      process.exit(0);
> +    return {};
>
> I'm not sure this will return...
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/johnuphoff/mailed-it/pull/2?email_source=notifications&email_token=AGBPQT3ZJIYSOC3723I5WELQK7WLNA5CNFSM4IZE5DR2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCFQGOMY#pullrequestreview-291530547>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGBPQT6JYUTAMYPXCCXPX5LQK7WLNANCNFSM4IZE5DRQ>
> .
>

lucask42 commented:
This all stemmed from the fact that `process.exit()` interrupts the testing.  In order to prevent the interruption I had created a mock function that did a noop.  I explored options to end the test early, force the test to pass,  or throw from within the mocked function but those strategies didn't seem to be viable.

In this change I just I removed the `process.exit()` and replaced it with `return {};`

This makes sense because `validAruments()` should return whatever the validArgs are and if there are no valid args or if the command is missing then `validAruments()` should just return an empty object.  This instead of returning a valid args OR killing the process.

Also the `process.exit()` in `validAruments()` is redundant.   When `template.js` gets arguments that have invalid name or invalid type it will go ahead and `process.exit()` and if it gets an unrecognized command it will just noop.

This kind of kicks the can down the road as there may be some more changes to write unit tests for `template.js` that can handle the `process.exit()`s there.

johnuphoff commented:
Cool yeah thanks - you're right that validArguments should return an empty object there.
